junit-jupiter-engine/src/test/java/org/junit/jupiter/api/AssertEqualsAssertionsTests.java

```diff
@@ -619,6 +625,50 @@ void assertEqualsWithUnequalObjectWhoseToStringImplementationThrowsAnException()
 		}
 	}
 
+	@Test
+	void assertEqualsWithBinaryPredicateBigDecimal() {
+		assertEquals(BigDecimal.valueOf(0.3), BigDecimal.valueOf(1.3).subtract(BigDecimal.valueOf(1.0)),
+			BinaryPredicate.compare());
+	}
+
+	@Test
+	void assertEqualsWithBinaryPredicateBigDecimalFail() {
+		try {
```

Comment:
I think in this cases you can use `org.assertj.core.api.Assertions.assertThatThrownBy`